### ISSUE TITLE
Bug: [unbound-method] Should not err in a `map` call when `thisArg` is passed

### ISSUE THREAD

**Issue by boris-petrov** [typescript-eslint/typescript-eslint#10931]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [x] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=5.8.2&fileType=.tsx&code=MYGwhgzhAECC0G8BQ1pmgXmgRgNxJWgA8AKASkUNQCcBTAFwFdqA7aegCwEsIA6MfKgC%2BSEUmAB7FhHppM0FrQDuccvgDa2ADTQATDoDMAXV4BbMAAcSYXkR1gyuIA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6WJgIwHsOATWgFtE%2BABbc%2B6KImjRu0SODABfEMqA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false

### Repro Code

```TypeScript
class A {
  a = 1;

  x() {
    return this.a;
  }
}

const a = new A();
[1, 2, 3].map(a.x, a);
```

### ESLint Config

```javascript
module.exports = {
  parser: "@typescript-eslint/parser",
  rules: {
    "@typescript-eslint/unbound-method": "error"
  },
};
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    "strictNullChecks": true
  }
}
```

### Expected Result

No error.

### Actual Result

Error.

### Additional Info

`map` takes a `thisArg` argument which the method is bound to. So a warning should not be issued in this case.

**Comment by bradzacher** [comment]
Duplicate of #743
